synapse/storage/databases/main/stream.py

```diff
@@ -1073,10 +1073,10 @@ def get_all_new_events_stream_txn(
             "get_all_new_events_stream", get_all_new_events_stream_txn
         )
 
-        events = await self.get_events_as_list(
-            event_to_received_ts.keys(),
-            get_prev_content=get_prev_content,
+        event_entries = await self.get_events_from_cache_or_db(
+            event_to_received_ts.keys(), get_prev_content=get_prev_content
         )
```

Comment:
Argh. I think this is going to cause an unacceptable change of behaviour for application services, which also use this codepath, and do care about redactions. (https://spec.matrix.org/v1.3/application-service-api/#pushing-events doesn't say that explicitly, but it does list a `redacted_because` field, which implies that redacted events should be redacted.)

So I guess I can see two potential approaches:
 1. Remove the call to `get_events_as_list` from `get_all_new_events_stream`, and just have it return `upper_bound, event_to_received_ts`. Rename it accordingly. Leave it up to the caller to call `get_events_as_list` or `get_events_from_cache_or_db` appropriately.
 2. Rework `get_events_as_list` so that it doesn't do the redaction magic if `redact_behaviour==EventRedactBehaviour.as_is`, and continue to use `get_events_as_list` here. Allow `redact_behaviour` to be passed in to `get_all_new_events_stream`.

I think I would find the former much easier to understand.